dali/test/python/test_pipeline.py

```diff
@@ -1654,3 +1654,102 @@ def test_invoke_serialize_error_handling_string():
 @raises(TypeError, "*define_graph*callable*")
 def test_invoke_serialize_error_handling_not_string():
     _identity_pipe().serialize(42)
+
+
+def check_dtype_ndim(dali_pipeline, output_dtype, output_ndim):
+    def ndim_dtype_matches(test_value, ref_value):
+        ref_value = ref_value if type(ref_value) is list else [ref_value]
+        for tv, ref in zip(test_value, ref_value):
+            if ref is None:
+                if tv != -1:
+                    return False
+            else:
+                if tv != ref:
+                    return False
+        return True
+
+    import tempfile
+    with tempfile.NamedTemporaryFile() as f:
+        dali_pipeline.serialize(filename=f.name)
+        deserialized_pipeline = Pipeline.deserialize(filename=f.name)
+        deserialized_pipeline.build()
+        assert ndim_dtype_matches(deserialized_pipeline.output_ndim(), output_ndim), f"`output_ndim` is not serialized properly. {deserialized_pipeline.output_ndim()} vs {output_ndim}."
+        assert ndim_dtype_matches(deserialized_pipeline.output_dtype(), output_dtype), f"`output_dtype` is not serialized properly. {deserialized_pipeline.output_dtype()} vs {output_dtype}."
+        deserialized_pipeline.run()
+    dali_pipeline.build()
+    dali_pipeline.run()
+
+
+@raises(RuntimeError, glob="Inconsistent output description.*")
+def check_dtype_ndim_with_raise(dali_pipeline, output_dtype, output_ndim):
+    check_dtype_ndim(dali_pipeline, output_dtype, output_ndim)
+
+
+def test_one_output_dtype_ndim():
+    @pipeline_def
+    def pipe():
+        inputs, labels = fn.readers.file(
+            file_root=os.path.join(get_dali_extra_path(), 'db', 'single', 'jpeg'), name="Reader")
+        decoded = fn.decoders.image(inputs, device="mixed", output_type=types.RGB)
+        return decoded
+
+    def create_test_package(output_dtype=None, output_ndim=None):
+        return pipe(batch_size=1, num_threads=1, device_id=0, output_dtype=output_dtype,
+                    output_ndim=output_ndim), output_dtype, output_ndim
+
+    both_correct = create_test_package(output_dtype=[types.UINT8], output_ndim=[3])
+    ndim_correct_dtype_wildcard = create_test_package(output_dtype=[None], output_ndim=[3])
+    dtype_correct_ndim_wildcard = create_test_package(output_dtype=types.UINT8)
+    dtype_incorrect = create_test_package(output_dtype=[types.FLOAT], output_ndim=[3])
+    ndim_incorrect = create_test_package(output_dtype=types.UINT8, output_ndim=0)
+    both_correct_one_list = create_test_package(output_dtype=types.UINT8, output_ndim=[3])
+    too_many_dtypes = create_test_package(output_dtype=[types.UINT8, types.FLOAT])
+    correct_dtypes_but_too_many = create_test_package(output_dtype=[types.UINT8, types.UINT8])
+    correct_ndims_but_too_many = create_test_package(output_ndim=[3, 3])
+    all_wildcards = create_test_package()
```

Comment:
How about passing incorrect values, like output_dtype=[int] or output_ndim=[-5.0]. etc. Also the specific defaults, that is -1 and DALI_NO_TYPE?

Reply:
Done